test/test_torch.py

```diff
@@ -17224,6 +17224,76 @@ def test_reciprocal_complex_extremal(self, device, dtype):
 
         self.compare_with_numpy(torch.reciprocal, np.reciprocal, vals, device, dtype)
 
+    def test_copysign_bool(self, device):
+        # Promoted to float
+        # Only 0 and 1
+        x = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.bool)
+        y = torch.tensor([0, 1, 0, 1], device=device, dtype=torch.bool)
+        res = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.float)
+        self.assertEqual(torch.copysign(x, y), res)
+
+    def _test_copysign(self, device, dtype, result_dtype):
+        # Scalar
+        x = torch.tensor([-1, 0, -0, 1], dtype=dtype)
+        y = torch.tensor(-1.)
+        res = torch.tensor([-1, -0, -0, -1], dtype=result_dtype)
+        self.assertEqual(torch.copysign(x, y), res)
```

Comment:
Now the interesting thing about self.assertEqual is that `-0.0 == 0.0`! So it is difficult to test that torch.copysign(torch.tensor(0.0), -1) actually returns torch.tensor(-0.0).

To test that, we could use copysign twice:
```
self.assertEqual(torch.copysign(torch.tensor(1.0), torch.copysign(torch.tensor(0.0), -1)), torch.tensor(-1.0))

Reply:
That's a great idea!